x-pack/plugins/security_solution/server/lib/risk_engine/risk_engine_data_client.ts

```diff
@@ -0,0 +1,154 @@
+    if (this.writerCache.get(namespace)) {
+      return this.writerCache.get(namespace);
+    }
+
+    await this.initializeResources({ namespace });
+    return this.writerCache.get(namespace);
+  }
+
+  private async initialiseWriter(namespace: string) {
+    const writer: Writer = {
+      bulk: async () => {},
+    };
+    this.writerCache.set(namespace, writer);
+    return this.writerCache.get(namespace);
+  }
+
+  public async initializeResources({
+    namespace = DEFAULT_NAMESPACE_STRING,
+  }: InitializeRiskEngineResourcesOpts) {
+    try {
+      const esClient = await this.options.elasticsearchClientPromise;
+
+      const indexPatterns = getIndexPattern(namespace);
+
+      const indexMetadata: Metadata = {
+        kibana: {
+          version: this.options.kibanaVersion,
+        },
+        managed: true,
+        namespace,
+      };
+
+      await createOrUpdateIlmPolicy({
+        logger: this.options.logger,
+        esClient,
+        name: ilmPolicyName,
+        policy: {
+          _meta: {
+            managed: true,
+          },
+          phases: {
+            hot: {
+              actions: {
+                rollover: {
+                  max_age: '30d',
+                  max_primary_shard_size: '50gb',
+                },
+              },
+            },
+          },
+        },
+      });
+
+      await createOrUpdateComponentTemplate({
+        logger: this.options.logger,
+        esClient,
+        template: {
+          name: mappingComponentName,
+          _meta: {
+            managed: true,
+          },
+          template: {
+            settings: {},
+            mappings: mappingFromFieldMap(riskFieldMap, 'strict'),
+          },
+        } as ClusterPutComponentTemplateRequest,
+        totalFieldsLimit,
+      });
+
+      await createOrUpdateIndexTemplate({
+        logger: this.options.logger,
+        esClient,
+        template: {
+          name: indexPatterns.template,
+          body: {
+            data_stream: { hidden: true },
+            index_patterns: [indexPatterns.alias],
+            composed_of: [mappingComponentName],
+            template: {
+              settings: {
+                auto_expand_replicas: '0-1',
+                hidden: true,
+                'index.lifecycle': {
+                  name: ilmPolicyName,
+                },
+                'index.mapping.total_fields.limit': totalFieldsLimit,
+              },
+              mappings: {
+                dynamic: false,
+                _meta: indexMetadata,
+              },
+            },
+            _meta: indexMetadata,
+          },
+        },
+      });
+
+      await createConcreteWriteIndex({
+        logger: this.options.logger,
+        esClient,
+        totalFieldsLimit,
+        indexPatterns,
+      });
+
+      this.initialiseWriter(namespace);
+    } catch (error) {
+      this.options.logger.error(`Error initializing risk engine resources: ${error}`);
```

Comment:
Note: we should test this with common ES errors that might arise (if you haven't already); I've seen cases where a lot of information is lost by just `toString()`ing like this. I believe we have helpers to stringify errors for this exact situation.